docs/4.4/api-reference.md

```diff
@@ -11,57 +11,287 @@ Teleport is currently working on documenting our API.
 
         We are currently working on this project. If you have an API suggestion, [please complete our survey](https://docs.google.com/forms/d/1HPQu5Asg3lR0cu5crnLDhlvovGpFVIIbDMRvqclPhQg/edit).
 
-## Authentication
-In order to interact with the Access Request API, you will need to provision appropriate
+# gRPC APIs
+
+In most cases, you can interact with teleport using our cli tools, [tsh](cli-docs.md#tsh) and [tctl](cli-docs.md#tctl). However, there are some scenarios where you may need to interact with teleport programmatically. For this purpose, the same gRPC auth API that `tctl` and `tsh` use is available for direct usage.
+
+## GoLang Examples
+Below are some code examples that can be used with Teleport to perform a few key tasks.
+
+Before you begin: 
+
+- Install [Golang](https://golang.org/doc/install) 1.13+ and Setup Go Dev Environment
+- Have access to a Teleport Auth server ([quickstart](quickstart.md))
+- [go get](https://golang.org/pkg/cmd/go/internal/get) the following packages (as you need them)
```

Comment:
Instead of `go get`, can you give a sample of `go.mod` file to include, so we promote the best practices

Reply:
This go.mod would be 67 lines long (basically just slimmed down version of teleport's go.mod), not sure if it fits well into the docs, but I can try it.

Reply:
@benarent @russjones What do you think we should do here? I would like to add a short flow like:
```
go mod init
go get github.com/gravitational/teleport
```
But there are a few issues:
- Teleport's `go.mod` has several indirect/replace rules that are necessary to run the API, so simply running `go get github.com/gravitational/teleport` to get the API doesn't work. The easiest way I found to get the API working in a go module is to copy teleport's `go.mod` and then run `go mod tidy` to slim it down a bit. 
- Regardless of the method of setting up the go module, it retrieves Teleport version 4.3.7, which I assume is the latest full release. These docs largely depend on the changes that are coming in #4630, #4573, and a small change in this pr itself. I'm guessing these changes won't be live until 5.0. So these docs won't be in a working state until then. Or if it's possible to `go get teleport-5.0-beta`, that could work, but I couldn't figure that out.

It seems like maybe we're trying to do too much with these docs, and the whole setup section would be better suited just linking to the `go-client` example. I don't see the use of having them copy and paste the whole `go-client` folder together.

Reply:
I would add it 

> EDIT: it definitely simplifies it for the user even if it's a bit of an eyesore.

Add it in, and I can add a `mkdocs` module to collapse it (Note to self: Add in `- pymdownx.details`) 

>     * Teleport version 4.3.7, which I assume is the latest full release.

That's interesting as our current release is 4.4.1 

>  whole setup section would be better suited just linking to the `go-client` example
I would keep it pretty high level; we don't have to explain each step but it should be enough hints that a junior Go Lang dev will know what to do.